packages/frontend/src/components/create-campaign/form/kpi-step/index.tsx

```diff
@@ -0,0 +1,320 @@
+    onError,
+}: KpiStepProps) {
+    const t = useTranslations("newCampaign.form.kpi");
+    const [enabled, setEnabled] = useState(false);
+    const [boundsError, setBoundsError] = useState("");
+    const [feedback, setFeedback] = useState(false);
+    const [minimumPayoutPercentage, setMinimumPayoutPercentage] = useState(
+        kpiSpecification?.minimumPayoutPercentage || 0,
+    );
+    const [lowerUsdTargetRaw, setLowerUsdTargetRaw] = useState<
+        NumberFormatValues | undefined
+    >(() => {
+        if (kpiSpecification) {
+            return {
+                floatValue: kpiSpecification.goal.lowerUsdTarget,
+                formattedValue: kpiSpecification.goal.lowerUsdTarget.toString(),
+                value: kpiSpecification.goal.lowerUsdTarget.toString(),
+            };
+        }
+        return undefined;
+    });
+    const [upperUsdTargetRaw, setUpperUsdTargetRaw] = useState<
+        NumberFormatValues | undefined
+    >(() => {
+        if (kpiSpecification) {
+            return {
+                floatValue: kpiSpecification.goal.upperUsdTarget,
+                formattedValue: kpiSpecification.goal.upperUsdTarget.toString(),
+                value: kpiSpecification.goal.upperUsdTarget.toString(),
+            };
+        }
+        return undefined;
+    });
+
+    const prevKpiSpecification = usePrevious(kpiSpecification);
+
+    const totalRewardsUsdAmount = useMemo(() => {
+        if (!rewards) return 0;
+        let total = 0;
+        for (const reward of rewards) {
+            if (!reward.amount.usdValue) return 0;
+            total += reward.amount.usdValue;
+        }
+        return total;
+    }, [rewards]);
+
+    // this hooks is used to disable and close the step when
+    // the kpi specification gets disabled, after the campaign creation
+    useEffect(() => {
+        if (enabled && !!prevKpiSpecification && !kpiSpecification)
+            setEnabled(false);
+    }, [enabled, kpiSpecification, prevKpiSpecification]);
+
+    useEffect(() => {
+        if (!feedback) return;
+
+        const timeout = setTimeout(() => {
+            setFeedback(false);
+        }, 1500);
+
+        return () => clearTimeout(timeout);
+    }, [feedback]);
+
+    useEffect(() => {
+        if (enabled) return;
+        if (kpiSpecification) onKpiChange({ kpiSpecification: undefined });
+
+        setMinimumPayoutPercentage(0);
+        setLowerUsdTargetRaw(undefined);
+        setUpperUsdTargetRaw(undefined);
+        setBoundsError("");
+    }, [enabled, kpiSpecification, onKpiChange]);
+
+    useEffect(() => {
+        if (!lowerUsdTargetRaw && !upperUsdTargetRaw) {
+            setBoundsError("");
+            return;
+        }
+
+        if (
+            (!lowerUsdTargetRaw && upperUsdTargetRaw) ||
+            (!upperUsdTargetRaw && lowerUsdTargetRaw)
+        ) {
+            setBoundsError("errors.missing");
+            return;
+        }
+
+        if (!lowerUsdTargetRaw || !upperUsdTargetRaw) return;
```

Comment:
The condition at line `123` is used to clear the error if both bounds are empty, so it's different from this since if one of the values is missing we should keep the error state as is, sounds good?

Reply:
Sounds good, lesgo